chemprop/data/utils.py

```diff
@@ -656,7 +656,46 @@ def get_inequality_targets(path: str, target_columns: List[str] = None) -> List[
                 raise ValueError(f'A target value in csv file {path} contains both ">" and "<" symbols. Inequality targets must be on one edge and not express a range.')
 
     return gt_targets, lt_targets
+def weight_split(data: MoleculeDataset,
+                 bias: str,
+                 sizes: Tuple[float, float, float] = (0.8, 0.1, 0.1)) -> Tuple[MoleculeDataset, MoleculeDataset,
+                                                                             MoleculeDataset]:
+    """
+    Splits a MoleculeDataset by molecular weight so that the largest or smallest molecules are in the train set.
+
+    :param data: A MoleculeDataset.
+    :param bias: A string indicating whether to bias the split towards the "big" or "small" molecules.
+    :param sizes: A length-3 tuple with the proportions of data in the train, validation, and test sets.
+    :param seed: Random seed for shuffling.
+    :return: A tuple of MoleculeDataset objects containing the train, validation, and test splits of the data.
+    """
+    if not (len(sizes) == 3 and sum(sizes) == 1):
+        raise ValueError(f"Invalid train/val/test splits! got: {sizes}")
```

Comment:
test for approximate equality to 1 rather than exact equality-otherwise point errors can come up here

Reply:
I put everything in the split_data function.